src/propulsion/calculate_RPM.cpp

```diff
@@ -0,0 +1,149 @@
+/*
+ * Author: George Karabassis
+ * Organisation: HYPED
+ * Date: 30/3/2018
+ * Description:
+ *
+ *    Copyright 2018 HYPED
+ *    Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file
+ *    except in compliance with the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software distributed under
+ *    the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ *    either express or implied. See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include "propulsion/calculate_RPM.hpp"
+
+#include <math.h>
+#ifndef M_PI
+#define M_PI           3.14159265358979323846
+#endif
+
+#include <cstdint>
+#include <fstream>
+#include <string>
+#include <sstream>
+#include <vector>
+#include <algorithm>
+
+// #include "motor_control/communicator.hpp"
+#include "data/data.hpp"
+#include "utils/system.hpp"
+
+namespace hyped {
+
+using data::NavigationType;
+using utils::System;
+using utils::Timer;
+
+namespace motor_control {
+
+constexpr double  kHalbachRadius    = 0.148;
+const std::string kAccelerationData = "../BeagleBone_black/data/configuration/AccelerationSlip.txt";
+
+CalculateRPM::CalculateRPM(Logger& log): 
+    data_(data::Data::getInstance()),
+    log_(log),
+    motor_failure_(false),
+    slip_calculated_(false)
+{
+    state_      = data_.getStateMachineData();
+    motor_data_.module_status = data::ModuleStatus::kStart;
+    motor_data_.velocity_1 = 0;
+    motor_data_.velocity_2 = 0;
+    motor_data_.velocity_3 = 0;
+    motor_data_.velocity_4 = 0;
+    data_.setMotorData(motor_data_);
+    motor_velocity_ = {0, 0, 0, 0};
+    //communicator_ = new Communicator(log);
+}
+
+    void CalculateRPM::updateMotorFailure(){
+        log_.ERR("MOTOR", "Motor State: MOTOR FAILURE\n");
+        motor_data_ = data_.getMotorData();
+        motor_data_.module_status = data::ModuleStatus::kCriticalFailure;
+        data_.setMotorData(motor_data_);
+        motor_failure_ = true;
+    }
+
+    std::vector<std::vector<double>> CalculateRPM::transpose(std::vector<std::vector<double>> data){
+        std::vector<std::vector<double>> transpose(2, std::vector<double>(data.size(), 1));
+        for (uint16_t i = 0; i < data.size(); ++i) {
+            for (uint16_t j = 0; j < 2; ++j) {
+            transpose[j][i] = data[i][j];
+            }
+        }
+        return transpose; 
+    }
+
+    bool CalculateRPM::check_file(std::string filepath){
```

Comment:
provide a public initialize function that initializes the class and it should return true if everything worked fine or false if there was an error

Reply:
Could you please explain a bit more for this suggestion?

Reply:
you could implement a initialize function that basically loads the data from the file and then it returns true if everything worked or false if there was an error